### PR TITLE
QBL-2136:Added test cases for ProfilePlus connect flow CTA

### PR DESCRIPTION
**Pull request by mridula18**
Please help review test cases for Connect Flow CTA's

### CODE REVIEW

File: src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAProfilePlusTurnedOff.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		userLogin.logoutAfterException();
+		userLogin.signInFromLink(userAdmin.getUsername(), userAdmin.getPassword());
+		AdminProfilePlusHelper.enableProfilePlus();
+		userLogin.logout();
+	}
+	private void kloutConnectAction(){
+		profilePlusSettingsView.open();
+		profilePlusSettings.connectToEmail();
+		profilePlusSettingsView.open();
+		Assert.assertTrue("Connect flow not working.", profilePlusSettings.isStatusConnected());
```
Comment (JGarcia-li) [inline src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAProfilePlusTurnedOff.java]: Is this method used in multiple test cases or just one? I think the assertion should be explicitly placed in the test case, the actions leading up to it are fine where they are.

File: src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTARoleRestriction.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			userLogin.signInFromLink(user.getUsername(), user.getPassword());
+			ctaView.open();
+			Assert.assertFalse("CTA  displayed, role restriction no applied", ctaView.getCTAElement("stateOne").isDisplayed());
+			Assert.assertFalse("CTA  displayed, role restriction no applied", ctaView.getCTAElement("stateTwo").isDisplayed());
+			Assert.assertFalse("CTA  displayed, role restriction no applied", ctaView.getCTAElement("stateThree").isDisplayed());
+			userLogin.logout();
+			ctaView.open();
+			Assert.assertFalse("CTA  displayed for anonymous user, role restriction no applied", ctaView.getCTAElement("stateOne").isDisplayed());
+			Assert.assertFalse("CTA  displayed for anonymous user, role restriction no applied", ctaView.getCTAElement("stateTwo").isDisplayed());
+			Assert.assertFalse("CTA  displayed for anonymous user, role restriction no applied", ctaView.getCTAElement("stateThree").isDisplayed());
```
Comment (JGarcia-li) [inline src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTARoleRestriction.java]: consider breaking this test case up into tests for specific roles. While it's convenient to have it grouped, if the user can see the CTA displayed, but the anonymous user couldn't, we wouldn't know the anonymous user was successful and that could imply something odd in the restriction. Or is this a 1:1 correlation?  When the user fails should the anon fail as well?

File: src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAUserRestriction.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		userLogin.signInFromLink(user.getUsername(), user.getPassword());
+		ctaView.open();
+		Assert.assertFalse("CTA  displayed, role restriction no applied", ctaView.getCTAElement("stateOne").isDisplayed());
+		Assert.assertFalse("CTA  displayed, role restriction no applied", ctaView.getCTAElement("stateTwo").isDisplayed());
+		Assert.assertFalse("CTA  displayed, role restriction no applied", ctaView.getCTAElement("stateThree").isDisplayed());
+		userLogin.logout();
+		ctaView.open();
+		Assert.assertFalse("CTA  displayed for anonymous user, role restriction no applied", ctaView.getCTAElement("stateOne").isDisplayed());
+		Assert.assertFalse("CTA  displayed for anonymous user, role restriction no applied", ctaView.getCTAElement("stateTwo").isDisplayed());
+		Assert.assertFalse("CTA  displayed for anonymous user, role restriction no applied", ctaView.getCTAElement("stateThree").isDisplayed());
```
Comment (JGarcia-li) [inline src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAUserRestriction.java]: What's the difference between this method and the testCTAForRole method? They look confusingly similar.

File: src/main/java/com/lithium/mineraloil/lia/ProfilePlus/CTAView.java [hunk 4]
```diff
@@ -50,4 +58,25 @@ public boolean isSatisfied() {
 		}.setTimeout(TimeUnit.SECONDS, 240).setPollInterval(TimeUnit.SECONDS, 1).waitUntilSatisfied();
 		Assert.assertTrue("Personalized feed banner not exists", personalizedFeedView.getPesonalizedFeedBanner().isDisplayed());
 	}
+
+	public String getCTAText(String state){
+		return ElementFactory.createBaseElement(By.xpath("//div[contains(@ng-if,'"+state+"')]/h2")).getText();
```
Comment (JGarcia-li) [inline src/main/java/com/lithium/mineraloil/lia/ProfilePlus/CTAView.java]: Why do you need the getCTAText method when there's a getCTAElement method that you could call .getText() on (among other interactions)?

File: src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAForFacebookConnectedUser.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	private ProfilePlusSettingsView profilePlusSettingsView = new ProfilePlusSettingsView();
+	private ProfilePlusSettings profilePlusSettings = new ProfilePlusSettings();
+	private UserProfilePPView userProfilePPView = new UserProfilePPView();
+	@Test
+	public void checkCTAForFacebookConnectUser() {
+		User user = CreateUser.createVerifiedUserWith("test");
+		userLogin.signInFromLink(user.getUsername(),user.getPassword());
+		profilePlusSettingsView.open();
+		profilePlusSettingsView.getConnectLink().click();
+		profilePlusSettings.connectToFaceBook("dzbhuvrxwx_1482360238@tfbnw.net","p@ssword123");
```
Comment (mridula18) [inline src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAForFacebookConnectedUser.java]: I am also considering this change but i am yet to decide a suitable place to add it to . I will make the change once i have finalized all the test flows

File: src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAProfilePlusTurnedOff.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+		kloutConnectAction();
+		userLogin.logout();
+		userLogin.signInFromLink(userAdmin.getUsername(), userAdmin.getPassword());
+		AdminProfilePlusHelper.disableProfilePlus();
+		userLogin.logout();
+		userLogin.signInFromLink(user.getUsername(), user.getPassword());
+		ctaView.open();
+		Assert.assertFalse("CTA  displayed, role restriction no applied", ctaView.getCTAElement("stateOne").isDisplayed());
+		Assert.assertFalse("CTA  displayed, role restriction no applied", ctaView.getCTAElement("stateTwo").isDisplayed());
+		Assert.assertFalse("CTA  displayed, role restriction no applied", ctaView.getCTAElement("stateThree").isDisplayed());
```
Comment (mridula18) [inline src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAProfilePlusTurnedOff.java]: Fixed the typo on all files

File: src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAProfilePlusTurnedOff.java (hunk 1 above)
Comment (mridula18) [inline src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAProfilePlusTurnedOff.java]: Moved the assert back to the test method

File: src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTARoleRestriction.java (hunk 2 above)
Comment (mridula18) [inline src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTARoleRestriction.java]: if the user sees in any state other than the role restricted for, then i consider the test as a fail. That was my reasoning for grouping them all under one test.Let me know if it needs to be changed

File: src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAUserRestriction.java (hunk 3 above)
Comment (mridula18) [inline src/integration-test/java/com/lithium/mineraloil/lia/test/profileplus/CTAUserRestriction.java]: CTAforRole is restricting CTA to be displayed only for users within a certain role 
CTAforUser is restricting CTA to be displayed only for certain restricted users

File: src/main/java/com/lithium/mineraloil/lia/ProfilePlus/CTAView.java (hunk 4 above)
Comment (mridula18) [inline src/main/java/com/lithium/mineraloil/lia/ProfilePlus/CTAView.java]: Removed the unwanted getCTAText.

**Pull request merged**